sycl/source/detail/event_impl.hpp

```diff
@@ -256,6 +259,25 @@ class event_impl {
     return MContext;
   }
 
+  // Sets a sync point which is used when this event represents an enqueue to a
+  // Command Bufferr.
+  void setSyncPoint(sycl::detail::pi::PiExtSyncPoint SyncPoint) {
+    MSyncPoint = SyncPoint;
+  }
+
+  // Get the sync point associated with this event.
+  sycl::detail::pi::PiExtSyncPoint getSyncPoint() const { return MSyncPoint; }
+
+  void setCommandGraph(
+      std::shared_ptr<ext::oneapi::experimental::detail::graph_impl> Graph) {
+    MGraph = Graph;
+  }
+
+  std::shared_ptr<ext::oneapi::experimental::detail::graph_impl>
+  getCommandGraph() const {
+    return MGraph.lock();
```

Comment:
Not sure if we need extra checks here. Because if the managed object is empty an empty pointer is returned. And we check for empty pointer whenever this function is called.

Reply:
Fair point. I could just be left to the caller, assuming they are ready for that being a potential returned value.